cirq-core/cirq/transformers/merge_k_qubit_gates.py

```diff
@@ -0,0 +1,77 @@
+# Copyright 2022 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Transformer pass to merge connected components of k-qubit unitary operations."""
+
+from typing import cast, Optional, Callable, TYPE_CHECKING
+
+from cirq import ops, protocols, circuits
+from cirq.transformers import transformer_api, transformer_primitives
+
+if TYPE_CHECKING:
+    import cirq
+
+
+@transformer_api.transformer
+def merge_k_qubit_unitaries(
+    circuit: 'cirq.AbstractCircuit',
+    *,
+    context: Optional['cirq.TransformerContext'] = None,
+    k: int = 0,
+    rewriter: Optional[Callable[['cirq.CircuitOperation'], 'cirq.OP_TREE']] = None,
+) -> 'cirq.Circuit':
```

Comment:
I'm a little confused now, this seems like a kind of drastic API change. What is k for if it can only be zero or one ? This also errors by default ?

Reply:
`k` can by any integer > 0 . This will be useful to replace `MergeInteractions` and `MergeInteractionsToSqrtISwap`, both of which need to merge connected components of 2q gates. We can directly call this method with `k=2` instead of rolling out another implementation.